devtools/x-lint/src/file_path.rs

```diff
@@ -36,14 +37,14 @@ impl<'l> FilePathContext<'l> {
     }
 
     /// Returns the path of this file, relative to the root of the repository.
-    pub fn file_path(&self) -> &'l Path {
+    pub fn file_path(&self) -> &'l Utf8Path {
         &self.file_path
     }
 
     /// Returns the extension of the file. Returns `None` if there's no extension or if the
     /// extension isn't valid UTF-8.
     pub fn extension(&self) -> Option<&'l str> {
-        self.file_path.extension().map(OsStr::to_str).flatten()
+        self.file_path.extension()
```

Comment:
The comment and the invariant can probably be changed now?

Reply:
Ooh, will fix.

Reply:
Fixed.